core/tests/protractor/stateEditor.js

```diff
@@ -133,4 +135,22 @@ describe('State editor', function() {
 
     users.logout();
   });
+
+  it('should preserve input value when rule type changes', function() {
+    users.createUser('user4@example.com', 'user4');
+    users.login('user4@example.com');
+    workflow.createExploration('sums', 'maths');
+    editor.setContent(forms.toRichText('some content'));
+
+    editor.openInteraction('TextInput');
+    editor.customizeInteraction('TextInput', 'My PlaceHolder', 2);
+    editor.selectRuleInAddResponse('TextInput', 'Equals', false, 'Some Text');
+    editor.expectRuleParametersToBe('Some Text');
```

Comment:
The naming of this function should reflect that it runs within the response modal.